observatory-platform/observatory/platform/cli/generate_command.py

```diff
@@ -73,95 +66,140 @@ def generate_terraform_config(self, config_path: str):
+        # Add underscores between capitalised letters, make them all lowercase and strip underscores at the start/end
+        telescope_module = re.sub(r"([A-Z])", r"_\1", telescope_class).lower().strip("_")
 
-        # Render dag
-        dag = render_template(dag_path, telescope_module=telescope_module, telescope_name=telescope_name)
-
-        # Render telescope
-        telescope = render_template(telescope_path, telescope_name=telescope_name)
-
-        # Render test
-        test = render_template(test_path, telescope_name=telescope_name)
+        # Render templates
+        dag = render_template(
+            dag_path, telescope_module=telescope_module, telescope_class=telescope_class, author_name=author_name
+        )
+        telescope = render_template(
+            telescope_path, telescope_module=telescope_module, telescope_class=telescope_class, author_name=author_name
+        )
+        test = render_template(
+            test_path, telescope_module=telescope_module, telescope_class=telescope_class, author_name=author_name
+        )
+        doc = render_template(
+            doc_path,
+            telescope_module=telescope_module,
+            telescope_class=telescope_class,
+        )
+        schema = render_template(schema_path)
 
-        # Render documentation
-        doc = render_template(doc_path, telescope_name=telescope_name)
+        # Get paths to files
+        observatory_dir = module_file_path("observatory.platform", nav_back_steps=-4)
 
-        # Save templates
-        dag_dst_dir = observatory.dags.dags.__path__[0]
-        dag_dst_file = os.path.join(dag_dst_dir, telescope_file)
+        dag_dst_dir = module_file_path("observatory.dags.dags")
+        dag_dst_file = os.path.join(dag_dst_dir, f"{telescope_module}.py")
 
-        telescope_dst_dir = observatory.dags.telescopes.__path__[0]
-        telescope_dst_file = os.path.join(telescope_dst_dir, telescope_file)
+        telescope_dst_dir = module_file_path("observatory.dags.telescopes")
+        telescope_dst_file = os.path.join(telescope_dst_dir, f"{telescope_module}.py")
 
-        test_dst_dir = "tests/observatory/dags/telescopes"
+        test_dst_dir = os.path.join(observatory_dir, "tests", "observatory", "dags", "telescopes")
         test_dst_file = os.path.join(test_dst_dir, f"test_{telescope_module}.py")
 
-        doc_dst_dir = "docs"
-        doc_dst_file = os.path.join(doc_dst_dir, "telescopes", f"{telescope_module}.md")
+        doc_dst_dir = os.path.join(observatory_dir, "docs", "telescopes")
+        doc_dst_file = os.path.join(doc_dst_dir, f"{telescope_module}.md")
 
-        doc_index_file = os.path.join(doc_dst_dir, "telescopes", "index.rst")
+        schema_dst_dir = module_file_path("observatory.dags.database.schema")
+        schema_dst_file = os.path.join(schema_dst_dir, f"{telescope_module}_{datetime.now().strftime('%Y-%m-%d')}.json")
```

Comment:
I know that it was like this previously, however, the generation tool would not be very useful to users if all of the files can only be generated into the `observatory-dags` project. Also, the `observatory-platform` shouldn't need to depend on the `observatory-dags` project and `module_file_path` might find an installed version of the module you are looking for, not the path in the project you are working on.

The ideal behaviour would be:
* When a `generate telescope` command is run, it generates the files in the correct places within the project the command is being run within.
* When `generate telescope` is run outside of a project, it prints an error that it needs to be run within a project.
* `generate telescope` can be run from anywhere in the project?
* One shouldn't assume that the Python project the files are being generated for is installed.

I'm not sure what the best way is to find the Python package folder, e.g. `observatory-dags/observatory/dags`. Once we split the DAGs out of observatory-platform we can make them normal Python packages (not namespaced ones), in which case, there might be a standard way to find the Python package name and then generate the files inside there.